presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -211,6 +199,31 @@ else if (state.isDone()) {
         }
     }
 
+    private synchronized void registerDynamicFilteringQuery()
+    {
+        if (!isEnableDynamicFiltering(stateMachine.getSession())) {
+            return;
+        }
+
+        if (isDone()) {
+            // query already finished
+            return;
+        }
+
+        dynamicFilterService.registerQuery(this);
+    }
+
+    private synchronized void unregisterDynamicFilteringQuery()
+    {
+        if (!isEnableDynamicFiltering(stateMachine.getSession())) {
+            return;
+        }
+
+        if (isDone()) {
+            dynamicFilterService.removeQuery(stateMachine.getQueryId());
```

Comment:
Can we do this without the condition on `isEnableDynamicFiltering`?  If so, I would just for safety.